### PR TITLE
[enterprise-4.12]OCPBUGS#16845: allocateLoadBalancerNodePorts support release notes

### PR DESCRIPTION
**Pull request by snarayan-redhat**
Version(s): 4.12

Issue: https://issues.redhat.com/browse/OCPBUGS-16845

Link to docs preview: https://63650--docspreview.netlify.app/openshift-enterprise/latest/release_notes/ocp-4-12-release-notes#ocp-4-12-networking-networkapi

QE review:
- [x] QE has approved this change.
- [x] SME has approved this change

### CODE REVIEW

**Comment by andreaskaris** [comment]
Looks good to me but I'll let others comment, too

**Review by tssurya — COMMENTED** [review]
lgtm
minor recommendation would be to link to the actual upstream K8s docs for this field which clearly states the purpose of the field in better terminology and then users can just read that  for more info kinda of thing.

**Comment by snarayan-redhat** [comment]
> lgtm minor recommendation would be to link to the actual upstream K8s docs for this field which clearly states the purpose of the field in better terminology and then users can just read that for more info kinda of thing.

Thanks @tssurya!
Is this the link that you are referring to?
https://kubernetes.io/docs/concepts/services-networking/service/#load-balancer-nodeport-allocation

**Comment by anuragthehatter** [comment]
Looks clean to me. Thanks

/lgtm

File: release_notes/ocp-4-12-release-notes.adoc [hunk 1]
```diff
@@ -931,6 +931,12 @@ For more information, see xref:../networking/hardware_networks/switching-bf2-nic
 
 Previously, for clusters that use the xref:../networking/ovn_kubernetes_network_provider/about-ovn-kubernetes.adoc#about-ovn-kubernetes[OVN-Kubernetes network plugin], during cluster installation you could enable hybrid networking so that your cluster supported Windows nodes. Now you can enable hybrid networking after installation. For more information, see xref:../networking/ovn_kubernetes_network_provider/configuring-hybrid-networking.adoc#configuring-hybrid-networking[Configuring hybrid networking].
 
+[id="ocp-4-12-networking-networkapi"]
+==== Support for allocateLoadBalancerNodePorts in Service object of Network API
+The `ServiceSpec` component in Network API under the `Service` object describes the attributes that a user creates on a service. The `allocateLoadBalancerNodePorts` attribute within the `ServiceSpec` component is now supported from {product-title} 4.12.28 release onwards. The `allocateLoadBalancerNodePorts` attribute defines whether the `NodePorts` will be automatically allocated for services of the `LoadBalancer` type.
+
+For more information, see xref:../rest_api/network_apis/service-v1.adoc#spec[Network API ServiceSpec object]
```
Comment (agantony) [inline release_notes/ocp-4-12-release-notes.adoc:938]:
```suggestion
For more information, see xref:../rest_api/network_apis/service-v1.adoc#spec[Network API ServiceSpec object].
```

File: release_notes/ocp-4-12-release-notes.adoc [hunk 2]
```diff
@@ -931,6 +931,12 @@ For more information, see xref:../networking/hardware_networks/switching-bf2-nic
 
 Previously, for clusters that use the xref:../networking/ovn_kubernetes_network_provider/about-ovn-kubernetes.adoc#about-ovn-kubernetes[OVN-Kubernetes network plugin], during cluster installation you could enable hybrid networking so that your cluster supported Windows nodes. Now you can enable hybrid networking after installation. For more information, see xref:../networking/ovn_kubernetes_network_provider/configuring-hybrid-networking.adoc#configuring-hybrid-networking[Configuring hybrid networking].
 
+[id="ocp-4-12-networking-networkapi"]
+==== Support for allocateLoadBalancerNodePorts in Service object of Network API
+The `ServiceSpec` component in Network API under the `Service` object describes the attributes that a user creates on a service. The `allocateLoadBalancerNodePorts` attribute within the `ServiceSpec` component is now supported from {product-title} 4.12.28 release onwards. The `allocateLoadBalancerNodePorts` attribute defines whether the `NodePorts` will be automatically allocated for services of the `LoadBalancer` type.
```
Comment (agantony) [inline release_notes/ocp-4-12-release-notes.adoc:936]:
```suggestion
The `ServiceSpec` component in the Network API under the `Service` object describes the attributes that a user creates for a service. The `allocateLoadBalancerNodePorts` attribute within the `ServiceSpec` component is now supported as of {product-title} 4.12.28. The `allocateLoadBalancerNodePorts` attribute defines whether the `NodePorts` for `LoadBalancer` type services are automatically allocated.
```

File: release_notes/ocp-4-12-release-notes.adoc [hunk 3]
```diff
@@ -931,6 +931,12 @@ For more information, see xref:../networking/hardware_networks/switching-bf2-nic
 
 Previously, for clusters that use the xref:../networking/ovn_kubernetes_network_provider/about-ovn-kubernetes.adoc#about-ovn-kubernetes[OVN-Kubernetes network plugin], during cluster installation you could enable hybrid networking so that your cluster supported Windows nodes. Now you can enable hybrid networking after installation. For more information, see xref:../networking/ovn_kubernetes_network_provider/configuring-hybrid-networking.adoc#configuring-hybrid-networking[Configuring hybrid networking].
 
+[id="ocp-4-12-networking-networkapi"]
+==== Support for allocateLoadBalancerNodePorts in Service object of Network API
```
Comment (agantony) [inline release_notes/ocp-4-12-release-notes.adoc:935]:
```suggestion
==== Support for allocateLoadBalancerNodePorts in the Network API service object
```

**Review by agantony — COMMENTED** [review]
Added a few suggestions; lgtm

File: release_notes/ocp-4-12-release-notes.adoc (hunk 2 above)
Comment (snarayan-redhat) [reply release_notes/ocp-4-12-release-notes.adoc:936]: Thanks for the great comments, Agil!
For couple of suggestions above, I would keep it the same for consistency with the API docs [here](https://docs.openshift.com/container-platform/4.12/rest_api/network_apis/service-v1.html#spec).

File: release_notes/ocp-4-12-release-notes.adoc (hunk 1 above)
Comment (snarayan-redhat) [reply release_notes/ocp-4-12-release-notes.adoc:938]: Typo! Good catch.

**Pull request merged**